src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,381 @@
+
+            - matrix 1 is row-based.
+            - matrix 2 is column based.
+
+            Let's write a byte to both matrices and see what happens.
+
+            Byte: 00011001
+
+            The following is displayed -- "o" means lit; "x" is unlit.
+
+            *matrix 2*
+            oxxxx | oxxoo
+            xxxxx | xxxxx
+            xxxxx | xxxxx
+            oxxxx | xxxxx
+            oxxxx | xxxxx
+            xxxxx | xxxxx
+            xxxxx | xxxxx
+            x       x
+                    *matrix 1*
+
+            If you write two bytes, then you'll write to the first two rows or
+            columns, respectively. With the matrix two, you can only write 5 bytes
+            and with matrix one, you can write 7, however, the number of bits that
+            are used differs.
+
+            Straightforwardly, both matrices start in the top-left, however
+            matrix 1 is row-based, left to right and matrix 2 is column-based
+            up to down.
+
+            You will notice these matrices are not symmetrical when you consider
+            rows vs columns. Theses matrices are 5x7 not 5x5 or 7x7.
+
+            That means that writing a byte -- 1000_0000 -- with just the high bit set
+            won't do anything for either matrix since that bit effectively "falls off".
+
+            Does this matter? Sorta.
+
+            If you are updating one pixel at a time, then you just need to update the
+            correct bit, one at a time, either row- or column-based (as you see in
+            the code below).
+
+            There are those extra pixels at the bottom, one per matrix. They are addressed
+            in the following way. Assume, you have a buffer defined like:
+
+            byte[] buffer = new byte[8]
+
+            To set the extra pixel, ensure the following bit is set:
+
+            - For Matrix 1: buffer[6] = 128;
+            - For Matrix 2: buffer[7] = 64;
+
+            buffer[6] is also used for the 7th row, on matrix 1. If you want to light up all
+            pixels and the special pixel, then set that buffer value to 159. That's all the bits but two of the highest ones (32 and 64).
+
+            If you have a bitmap you want to write all at once, then you can simply write
+            rows of bytes to matrix 1. However, you need to ensure your bitmap only uses
+            the first 5 bits. If you are writing to matrix 2 then writing one row at a
+            time won't work. Your content will be 90d flipped and the content (if you write
+            the same 5-bit encoded bitmap) will be 2 pixels short. Instead, you need to
+            translate row-based content to columns, similar to the `else` clause below.
+
+            Separately, you can disable one matrix or the other. That's largely a separate
+            concern. It has nothing to do with byte structure.
+            */
+
+            int logicalRow = matrix is 0 ? y : x;
```

Comment:
in theory, you should check that matrix, x and y are in the proper range because here it can break if not in the range.
Same for couple of other functions below